python/cudf/cudf/core/series.py

```diff
@@ -227,9 +227,10 @@ def __setitem__(self, key, value):
                     value.dtype, self._frame._column.dtype
                 )
                 value = value.astype(to_dtype)
-                self._frame._column._mimic_inplace(
-                    self._frame._column.astype(to_dtype), inplace=True
-                )
+                if self._frame._column.dtype != to_dtype:
+                    self._frame._column._mimic_inplace(
+                        self._frame._column.astype(to_dtype), inplace=True
+                    )
```

Comment:
Rationale for this change is that `mimic_inplace` is not required if the dtypes match.

Reply:
Yup